lib/video_info/providers/vkontakte.rb

```diff
@@ -51,19 +61,38 @@ def duration
 
       private
 
+      def _make_request(url, options)
+        request = Net::HTTP::Post.new(url.path)
+        request.body = URI.encode_www_form(options)
+        conn = Net::HTTP.new(url.host, url.port)
+        conn.use_ssl = true
+        resp = conn.request(request)
+        resp.body.force_encoding('cp1251').encode('UTF-8', undef: :replace)
+      end
+
       def _set_data_from_api
-        uri = open(url, options)
-        if RUBY_VERSION.to_i < 2
-          Iconv.iconv('utf-8', 'cp1251', uri.read)[0]
-        else
-          uri.read.encode('UTF-8')
-        end
+        url = URI('https://vk.com/al_video.php')
+        options['act'] = 'show'
+        options['al'] = '1'
+        options['video'] = "#{@video_owner}_#{@video_id}"
+        _make_request(url, options)
+      end
+
+      def _error_found?(response)
+        title = response.body[/<title>(.*)<\/title>/, 1]
+                        .force_encoding('cp1251')
+                        .encode('UTF-8', undef: :replace)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.